packages/components/src/color-palette/utils.ts

```diff
@@ -27,9 +27,11 @@ export const extractColorNameFromCurrentValue = (
 	}
 
 	const currentValueIsCssVariable = /^var\(/.test( currentValue );
-	const normalizedCurrentValue = currentValueIsCssVariable
-		? currentValue
-		: colord( currentValue ).toHex();
+	const currentValueIsColorMix = /^color-mix\(.*\)$/.test( currentValue );
```

Comment:
I'd apply a less strict regex here for better matching:

```suggestion
	const currentValueIsColorMix = /color-mix\(/.test( currentValue );
```

Reply:
And potentially consider removing the "start of line" caret (`^`) from `currentValueIsCssVariable`  too